internal/keyvault/authorizer.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keyvault
+
+import (
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/adal"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+// GetKeyvaultAuthorizer prepares a specifc authorizer for keyvault use
+func GetKeyvaultAuthorizer() (autorest.Authorizer, error) {
+	azureEnv, _ := azure.EnvironmentFromName("AzurePublicCloud")
+	vaultEndpoint := strings.TrimSuffix(azureEnv.KeyVaultEndpoint, "/")
+	tenant := os.Getenv("AZURE_TENANT_ID")
+	clientID := os.Getenv("AZURE_CLIENT_ID")
+	clientSecret := os.Getenv("AZURE_CLIENT_SECRET")
+
+	alternateEndpoint, _ := url.Parse(
+		"https://login.windows.net/" + tenant + "/oauth2/token",
+	)
+
+	var a autorest.Authorizer
+	var err error
+
+	oauthconfig, err := adal.NewOAuthConfig(azureEnv.ActiveDirectoryEndpoint, tenant)
+	if err != nil {
+		return a, err
```

Comment:
Please wrap this error with some context and `%w`

Reply:
done